pandas/core/indexing.py

```diff
@@ -542,7 +542,18 @@ def can_do_equal_len():
                         is_integer(indexer[info_axis]) and
                         all(is_null_slice(idx) for i, idx in enumerate(indexer)
                             if i != info_axis)):
-                    self.obj[item_labels[indexer[info_axis]]] = value
+                    index = indexer[info_axis]
+                    target = self.obj[item_labels[index]]
+
+                    # Duplicate columns
```

Comment:
this is too much specific if thening. you are trying to catch a very specific case and not solving it in a more general way.